src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/reducers/__tests__/chartBuilderReducer.test.ts

```diff
@@ -966,5 +1152,371 @@ describe('chartBuilderReducer', () => {
+            {
+              dataSet: {
+                filters: ['ethnicity-major-chinese', 'state-funded-primary'],
+                indicator: 'authorised-absence-sessions',
+                timePeriod: '2015_AY',
+              },
+              dataGrouping: { type: 'EqualIntervals', customGroups: [] },
+              boundaryLevel: 2,
+            },
+            {
+              dataSet: {
+                filters: ['ethnicity-major-chinese', 'state-funded-primary'],
+                indicator: 'authorised-absence-sessions',
+                timePeriod: '2014_AY',
+              },
+              dataGrouping: { type: 'Custom', customGroups: [] },
+              boundaryLevel: 3,
+            },
+          ],
+        },
+      };
+      const expectedChartBuilderStateProps: Omit<ChartBuilderState, 'map'> = {
+        axes: {
+          major: {
+            type: 'major',
+            groupBy: 'locations',
+            dataSets: [
+              {
+                filters: ['ethnicity-major-chinese', 'state-funded-primary'],
+                indicator: 'authorised-absence-sessions',
+                timePeriod: '2015_AY',
+              },
+              {
+                filters: ['ethnicity-major-chinese', 'state-funded-primary'],
+                indicator: 'authorised-absence-sessions',
+                timePeriod: '2014_AY',
+              },
+            ],
+            referenceLines: [],
+            visible: true,
+            label: {
+              text: '',
+            },
+          },
+        },
+        definition: mapBlockDefinition,
+        legend: {
+          position: 'top',
+          items: [],
+        },
+        options: {
+          alt: '',
+          height: 300,
+          subtitle: '',
+          title: 'Chart title',
+          titleType: 'alternative',
+          boundaryLevel: 10,
+        },
+      };
+
+      const { result, rerender } = renderHook(() =>
+        useChartBuilderReducer({
+          data: testFullTable.results,
+          meta: testFullTable.subjectMeta,
+          tableTitle: 'Table title',
+          chart: initialConfiguration,
+        }),
+      );
+
+      result.current.actions.updateMapBoundaryLevels({
+        boundaryLevel: 10,
+        dataSetConfigs: [
+          {
+            dataSet: {
+              filters: ['ethnicity-major-chinese', 'state-funded-primary'],
+              indicator: 'authorised-absence-sessions',
+              timePeriod: '2015_AY',
+            },
+            boundaryLevel: 20,
+          },
+          {
+            dataSet: {
+              filters: ['ethnicity-major-chinese', 'state-funded-primary'],
+              indicator: 'authorised-absence-sessions',
+              timePeriod: '2014_AY',
+            },
+            boundaryLevel: 30,
+          },
+        ],
+      });
+      rerender();
+
+      expect(result.current.state).toEqual<ChartBuilderState>({
+        ...expectedChartBuilderStateProps,
+        map: {
```

Comment:
Would suggest to simplify this test and make it quicker to read, we don't re-check the entire chart builder state.

We can just check the specific parts of the state we're actually interested in:

```ts
expect(result.current.state.options.boundaryLevel).toBe(...);
expect(result.current.state.map).toEqual<MapConfig>({
  // ...
});
```